File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.java
import java.util.List;
import java.util.ArrayList;
import java.util.Map;
import java.util.HashMap;


class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        
        List<String> possibles = new ArrayList<>();
        possibles.add("aya");
        possibles.add("ye");
        possibles.add("woo");
        possibles.add("ma");
        
        Map<String, Integer> map = Map.of(
            "a",3, 
            "y",2,
            "w",3,
            "m",2);
        
        for(String next : babbling){
            
            int removed = 0;
            for(String possible : possibles){
                
                if(next.contains(possible)){
                    removed+=possible.length();
                }
                
            }
            
            if(removed == next.length()) answer++;
        }
        
                
        return answer;
        
        
    }
}

File: 프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.py
def solution(babbling):
    answer = 0
    
    for i in range(len(babbling)):
        babbling[i] = babbling[i].replace("aya", "1")
        babbling[i] = babbling[i].replace("ye", "1")
        babbling[i] = babbling[i].replace("woo", "1")
        babbling[i] = babbling[i].replace("ma", "1")
        babbling[i] = babbling[i].replace("1", "")
        
        if len(babbling[i]) == 0:answer+=1
    
    return answer